dev/puppet-server.conf.sample

```diff
@@ -65,6 +65,16 @@ http-client: {
     #                TLS_RSA_WITH_AES_256_CBC_SHA,
     #                TLS_RSA_WITH_AES_128_CBC_SHA256,
     #                TLS_RSA_WITH_AES_128_CBC_SHA]
+
+    # The amount of time, in milliseconds, that  an outbound HTTP connection
+    # will wait for no data to be available before closing the socket. If not
+    # defined, defaults to 20 minutes.
+    #socket-timeout: 1200000
+
+    # The mount of time, in milliseconds, that an output HTTP connection will
+    # wait to connect to connect before giving up. It not defined, defaults to
+    # 20 minutes.
+    #connect-timeout: 1200000
```

Comment:
I'd prefer that we use a default of 2 minutes here since it aligns better with Puppet's client timeout.

I think connection timeouts should generally be much shorter than socket timeouts.  Socket timeouts can relatively be more damaging.  In the socket timeout case, the server might go and and perform the request it was asked to do even if the client gave up on getting the response back after some period of time.  In the connect timeout case, however, the server wouldn't have even gotten to the point of starting to act on the client's request so there's nothing to abort on the server side.